app/models/container_deployment.rb

```diff
@@ -0,0 +1,10 @@
+class ContainerDeployment < ApplicationRecord
+  has_many :container_deployment_authentications, :dependent => :destroy
+  belongs_to :deployed_ems, :class_name => 'ExtManagementSystem', :inverse_of => :container_deployments
+  belongs_to :deployed_on_ems, :class_name => 'ExtManagementSystem', :inverse_of => :container_deployments
```

Comment:
@alongoldboim after figuring out a good name :smile: I'd say you also put two `has_one` associations on the other side